### PR TITLE
Allow learning_rate=0

### PR DESCRIPTION
**Pull request by drasmuss**
**Motivation and context:**
Often when working with a learning model it is helpful to quickly disable learning by setting the learning rate to 0.  Currently we don't allow this in Nengo, which isn't the end of the world but just makes things a little awkward.  I think there is pretty minimal downside to allowing it; it seems unlikely that users will accidentally set the learning rate to zero when they do actually want learning.

**How long should this take to review?**
- Quick (less than 40 lines changed or changes are straightforward)

**Types of changes:**
- New feature (non-breaking change which adds functionality)

**Checklist:**

- [x] I have read the **CONTRIBUTING.rst** document.
- [n/a] I have updated the documentation accordingly.
- [n/a] I have included a changelog entry.
- [n/a] I have added tests to cover my changes.
- [n/a] All new and existing tests passed.

### CODE REVIEW

**Comment by studywolf** [comment]
this has come up for me a bunch of times would be very handy
